channeldb/kvdb/backend.go

```diff
@@ -25,28 +32,93 @@ func GetBoltBackend(path, name string, noFreeListSync,
 	autoCompact bool) (Backend, error) {
 
 	dbFilePath := filepath.Join(path, name)
-	var (
-		db  Backend
-		err error
-	)
 
+	// Is this a new database?
 	if !fileExists(dbFilePath) {
 		if !fileExists(path) {
 			if err := os.MkdirAll(path, 0700); err != nil {
 				return nil, err
 			}
 		}
 
-		db, err = Create(BoltBackendName, dbFilePath, noFreeListSync)
-	} else {
-		db, err = Open(BoltBackendName, dbFilePath, noFreeListSync)
+		return Create(BoltBackendName, dbFilePath, noFreeListSync)
+	}
+
+	// This is an existing database. We might want to compact it on startup
+	// to free up some space.
+	if autoCompact {
+		if err := compactAndSwap(path, name); err != nil {
```

Comment:
Non-blocking, but we'll eventually also want to start compacting the Sphinx replay database as well since it constantly has items removed from it each block.

Reply:
Makes sense, done.